File: ingestor/mongo_client.py
from pymongo import MongoClient
from dotenv import load_dotenv
from main import TIMESTAMP_KEY, METADATA_KEY
import os
import logging

load_dotenv()
logging.basicConfig(level=logging.INFO)

uri = "mongodb://{username}:{password}@{host}:{port}".format(
    username=os.getenv("MONGO_USER"),
    password=os.getenv("MONGO_PASSWORD"),
    host=os.getenv("MONGO_HOST"),
    port=os.getenv("MONGO_PORT"),
)

COLLECTION_NAME = "repos"

def get_mongo_client():
    return MongoClient(uri)

def create_collection(client):
    try:
        db = client.test
        collections = db.list_collection_names()
        if COLLECTION_NAME in collections:
            logging.info("collection already exists, skippping creation")
            return

        db.create_collection(
            COLLECTION_NAME,
            timeseries={
                "timeField": "ts",
                # "metaField": "u",
                "granularity": "hours",
            },
        )
    except Exception as e:
        logging.error("Error creating collection" + str(e))


def insert_documents(client, documents):
    try:
        db = client.test
        db[COLLECTION_NAME].insert_many(documents)
    except Exception as e:
        logging.error("Error inserting documents" + str(e))

File: ingestor-v2/src/mongoClient.ts
import { MongoClient } from "mongodb";
import { config } from "dotenv";
config();

const { MONGO_USER, MONGO_PASSWORD, MONGO_HOST, MONGO_PORT } = process.env;
const url = `mongodb://${MONGO_USER}:${MONGO_PASSWORD}@${MONGO_HOST}:${MONGO_PORT}`;

const COLLECTION_NAME = "repos";

export const getMongoClient = () => {
  return new MongoClient(url);
};

export const createTimeSeriesCollection = async (client: MongoClient) => {
  const db = client.db("test");

  const collections = await (
    await db.listCollections().toArray()
  ).map(({ name }) => name);
  const exists = collections.find((col) => col === COLLECTION_NAME);
  if (exists) {
    return;
  }

  await db.createCollection(COLLECTION_NAME, {
    timeseries: {
      timeField: "ts",
      granularity: "hours",
    },
    expireAfterSeconds: 30 * 24 * 60 * 60,
  });
};

export const insertDocuments = async (
  client: MongoClient,
  documents: Document[],
) => {
  const db = client.db("test");
  await db.collection(COLLECTION_NAME).insertMany(documents);
};
